### PR TITLE
Kernel: Always recompile silently methods

### PR DESCRIPTION
**Pull request by jecisc**
While updating a class, it migrates all the methods from the old class to the new one. In that case, the methods are added silently. Well... Usually. 

In the case of a class that has a trait, the methods on the class side are not added silently but with MethodAdded announcements. 

This change removes the annoucement to always do things silently while recompiling

### CODE REVIEW

**Comment by Ducasse** [comment]
@tesonep Pablo we should explain you a crazy bug I got. It was massive.

File: src/Flashback-Decompiler/FBDDecompiler.class.st [hunk 1]
```diff
@@ -577,61 +579,61 @@ FBDDecompiler >> quickMethod [
...
-		compile.	"Assume OK after proceed from SyntaxError"
-	selector == newMethod selector
-		ifFalse: [ self error: 'selector changed!' ].
-	oldClass 	addSelector: selector withRecompiledMethod: newMethod
+		             source: (self decompile: oldClass >> selector) formattedCode;
+		             class: oldClass;
+		             failBlock: [ ^ self ];
+		             compile. "Assume OK after proceed from SyntaxError"
+	selector == newMethod selector ifFalse: [ self error: 'selector changed!' ].
+	oldClass addSelectorSilently: selector withMethod: newMethod
```
Comment (jecisc) [inline src/Flashback-Decompiler/FBDDecompiler.class.st:592]: Change here

File: src/Kernel/Behavior.class.st [hunk 2]
```diff
@@ -71,11 +71,6 @@ Behavior >> addSelector: selector withMethod: compiledMethod [
 	^ self addSelectorSilently: selector withMethod: compiledMethod
 ]
 
-{ #category : 'adding-removing methods' }
-Behavior >> addSelector: selector withRecompiledMethod: compiledMethod [
```
Comment (jecisc) [inline src/Kernel/Behavior.class.st:75]: change here

File: src/OpalCompiler-Core/Behavior.extension.st [hunk 3]
```diff
@@ -58,45 +58,45 @@ Behavior >> compiler [
...
 
-{ #category : #'*OpalCompiler-Core' }
+{ #category : '*OpalCompiler-Core' }
 Behavior >> recompile: selector from: oldClass [
 	"Compile the method associated with selector in the receiver's method dictionary."
 
 	| newMethod |
 	newMethod := self recompileBasic: selector from: oldClass.
-	self addSelector: selector withRecompiledMethod: newMethod
+	self addSelectorSilently: selector withMethod: newMethod
```
Comment (jecisc) [inline src/OpalCompiler-Core/Behavior.extension.st:96]: Change here

File: src/TraitsV2/TraitedClass.class.st [hunk 4]
```diff
@@ -84,21 +86,12 @@ TraitedClass >> addAndClassifySelector: selector withMethod: compiledMethod inPr
 	TraitChange addSelector: selector on: self
 ]
 
-{ #category : #'accessing - method dictionary' }
-TraitedClass >> addSelector: selector withRecompiledMethod: compiledMethod [
```
Comment (jecisc) [inline src/TraitsV2/TraitedClass.class.st:88]: Change here

File: src/TraitsV2/TraitedMetaclass.class.st [hunk 5]
```diff
@@ -34,58 +36,50 @@ TraitedMetaclass >> addAndClassifySelector: selector withMethod: compiledMethod
 	TraitChange addSelector: selector on: self
 ]
 
-{ #category : #'accessing - method dictionary' }
-TraitedMetaclass >> addSelector: selector withRecompiledMethod: compiledMethod [
```
Comment (jecisc) [inline src/TraitsV2/TraitedMetaclass.class.st:38]: Change here

**Review by jecisc — COMMENTED** [review]
Here are all places in the code that changed

**Review by Gabriel-Darbord — APPROVED** [review]

**Pull request merged**